### ISSUE TITLE
Mention that Landscape is running in Urbit boot messages

### ISSUE THREAD

**Issue by msutherl** [urbit/urbit#1332]
Urbit currently doesn't give any indication that Landscape/Modulo exists. Wondering if it would be reasonable to include a message at the end of the boot sequence that says something like:

`Modulo running on https://localhost:[port]`

I suppose the most elegant way to do this would be from the Modulo app itself.

**Comment by ixv** [comment]
Sure thats definitely possible. It would be somewhat redundant with the `http: live (insecure, public) on 8080` print though.

**Comment by joemfb** [comment]
> activated app home/modulo

*technically speaking* ...

**Comment by vvisigoth** [comment]
AKSHUALLY

**Comment by msutherl** [comment]
@ixv maybe you're right. For consistency – or perhaps grotesque inconsistency – could be something like:

```
http: live (insecure, public) on 8080
modulo: live on localhost:8080
```

But there's something gross about that.

@joemfb perhaps the **right way** to do this would be to let each app publish its own "I'm activated" message.

```
activated app home/modulo: live on localhost:8080 🍅
```

**Comment by joemfb** [comment]
yikes

**Comment by msutherl** [comment]
@joemfb wait, you're right – that would be awful. The more I think about this the more I think apps shouldn't do this, even though it causes a temporary annoying UX gap.

**Comment by joemfb** [comment]
It's just that the thing that's started on 8080 or whatever is a general-purpose HTTP server, those "activated" messages tell you what applications are automatically started for you, etc. It doesn't feel right to specialize those basic system notifications for onboarding.

At the same time. I do think that urbit needs some kind of first-boot wizard/tour/etc. The UX is not friendly, and there should be some way to direct people to the happy path.

**Comment by msutherl** [comment]
@joemfb I had immediately dismissed the idea as unpalatable, but with your blessing, a dedicated "welcome" message that can easily be disabled by the purist would be great. Where would one ideally insert such a thing?

**Comment by joemfb** [comment]
The fully general answer to this is actually really hard. I talked to @galenwp about this awhile ago, but haven't had a chance to circle back. There isn't any central orchestration of all the layers of the system, so it's kind of hard to even answer when "boot" is complete (for some expansive definition of "boot"). But this doesn't to be fully general.

I'd say that if the message is short it can be completely in the console. If it's long, I'd make it available from the login page (maybe with associated server-side state to track if it's been dismissed, etc.), and just have a short message in the console directing the user to the longer one.

The modulo app is probably as good as a place as any to put such a console message. It would probably need to be printed after a timer to make sure that the initial (local) clay syncs complete first.

As for disabling, that's also hard to see. A straightforward option would be to disable the message for fake ships. It's currently not easy to tell if you're a fake ship, but that could be changed.

All this is kind of vague. Do you have such a welcome message in mind?

**Comment by philipcmonk** [comment]
Would be kind of nice to get a notification whenever an app binds a route,
to suggest where on the webserver you should look.

**Comment by vvisigoth** [comment]
Yes, theoretically, you could add a printf when chat binds, notes binds,
etc.?
—
~poldec-tonteg
http://urbit.org

**Comment by galenwp** [comment]
I think

```
http: live (insecure, public) on 8080
modulo: live on localhost:8080
```

is way too cryptic for an ordinary user. We want to tell them something like

```
Web interface available at localhost:8080 or ship.urbit.org
```

That copy probably isn't perfect, but I think it should be a one-liner that's a complete sentence and is easy to read and understand.

I'm not particularly opinionated about where this gets printed. Seems fine to have something related to Modulo do it if there's a concern about the statement being technically correct.

**Comment by ohAitch** [comment]
Probably want http:// and https:// respectively on those?

**Comment by galenwp** [comment]
Sure, sure.

**Comment by loganallenc** [comment]
Ping.

**Comment by matildepark** [comment]
@Fang- Can I assign this to you to just include a printf based on @galenwp's comment?

**Comment by Fang-** [comment]
I can probably do this for "localhost" urls. I'm not immediately sure if "ship.arvo.network" is possible though, would need to dive deep into http.c, eyre and dns again.

**Comment by galenwp** [comment]
My suggestion was just to say 'or' to leave it to the user to figure it out. It's meant to just be a suggestion.

**Comment by matildepark** [comment]
A pseudo-fix, the best of all fixes, in #3092.
